pkg/catalog/types.go

```diff
@@ -239,12 +239,53 @@ const (
 	BLOCKMETA_SEGID_IDX      = 6
 )
 
+type ObjectLocation [objectio.LocationLen]byte
+
+// ProtoSize is used by gogoproto.
+func (m *ObjectLocation) ProtoSize() int {
+	return objectio.LocationLen
+}
+
+// MarshalTo is used by gogoproto.
+func (m *ObjectLocation) MarshalTo(data []byte) (int, error) {
+	size := m.ProtoSize()
+	return m.MarshalToSizedBuffer(data[:size])
+}
+
+// MarshalToSizedBuffer is used by gogoproto.
+func (m *ObjectLocation) MarshalToSizedBuffer(data []byte) (int, error) {
+	if len(data) < m.ProtoSize() {
+		panic("invalid byte slice")
+	}
+	n := copy(data, m[:])
+	return n, nil
+}
+
+// Marshal is used by gogoproto.
+func (m *ObjectLocation) Marshal() ([]byte, error) {
+	data := make([]byte, m.ProtoSize())
+	n, err := m.MarshalToSizedBuffer(data)
+	if err != nil {
+		return nil, err
+	}
+	return data[:n], err
+}
+
+// Unmarshal is used by gogoproto.
+func (m *ObjectLocation) Unmarshal(data []byte) error {
+	if len(data) < m.ProtoSize() {
+		panic("invalid byte slice")
+	}
+	copy(m[:], data)
+	return nil
+}
+
 type BlockInfo struct {
 	BlockID    types.Blockid
 	EntryState bool
 	Sorted     bool
```

Comment:
blockInfo should be handled in a manner similar to a strong pointer rotation rather than a sequence talk.

Reply:
This PB formatting is required by [github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.BlockMeta](https://github.com/cnutshell/matrixone/blob/2c800c32bdee5da737dd50c16607548c364e2bce/pkg/vm/engine/disttae/types.go#L283).

And to be clear, this PR mainly replaces gob with PB or manual formatting.
Further optimization deserves another individual PR.

Reply:
The main is now a pointer operation, not a gob or a proto